kit/layout/src/Portal.tsx

```diff
@@ -71,22 +76,144 @@ type PortalItem = {
+                {children}
+            </View>
+        );
+    }
+
+    return <>{children}</>;
+});
+
+type PortalsViewProps = {
+    renderOnlyLastPortal: boolean;
+    // Sorted by priority array of portals
+    portalKeys: string[];
+    portals: { [key: string]: PortalItem | null };
+};
+
+const PortalsView = React.memo<PortalsViewProps>(function PortalsView({
+    renderOnlyLastPortal,
+    portalKeys,
+    portals,
+}) {
+    if (!renderOnlyLastPortal) {
+        return (
+            <>
+                {portalKeys.map(key => {
+                    const portal = portals[key];
+
+                    if (portal == null) {
+                        return null;
+                    }
+
+                    return <PortalView key={`portal_${key}`} {...portal} />;
+                })}
+            </>
+        );
+    }
+
+    const maxMountedKey = portalKeys[portalKeys.length - 1];
+
+    const lastPortal = portals[maxMountedKey];
+
+    if (lastPortal == null) {
+        return null;
+    }
+
+    return <PortalView key={`portal_${maxMountedKey}`} {...lastPortal} />;
+});
+
+const MaybeParentManager = React.memo(function MaybeParentManager({
+    setParentManager,
+}: {
+    setParentManager: (parentManager: PortalMethods) => void;
+}) {
+    const parentManager = React.useContext(PortalContext);
+
+    React.useEffect(() => {
+        if (parentManager == null) {
+            return;
+        }
+
+        setParentManager(parentManager);
+    }, [parentManager, setParentManager]);
+
+    return null;
+});
+
+type MaybePortalContentWrapperProps = {
+    wrapperComponent?: WrapperComponent | null;
+    children: React.ReactNode;
+    portals: React.ReactNode;
 };
 
+function MaybePortalContentWrapper({
+    wrapperComponent: ContentWrapperComponent,
+    children,
+    portals,
+}: MaybePortalContentWrapperProps) {
+    if (ContentWrapperComponent == null) {
+        return (
+            <>
+                {children}
+                {portals}
+            </>
+        );
+    }
+
+    return <ContentWrapperComponent portals={portals}>{children}</ContentWrapperComponent>;
+}
+
 export class PortalManager extends React.PureComponent<PortalManagerProps, PortalManagerState> {
-    state: PortalManagerState = {};
+    parentManager?: PortalMethods;
+
+    counter: number = 0;
+
+    manager: PortalMethods;
+
+    constructor(props: PortalManagerProps) {
+        super(props);
+
+        this.state = {};
+        this.manager = {
+            mount: this.mount,
+            update: this.update,
+            unmount: this.unmount,
+        };
+    }
 
     getMaxMountedKey() {
-        return Object.keys(this.state)
-            .sort((a, b) => Number(b) - Number(a)) // reversed order by keys
-            .find(key => !!this.state[Number(key)]); // find the first (max) key
+        const portals = this.state;
+        return (
+            Object.keys(portals)
+                .filter(portalKey => portals[Number(portalKey)] != null)
```

Comment:
Is it a sort of "bug fix"?)

Reply:
nah, I just re-shuffled it all a bit, and wrap Portals with memo :D